griptape/drivers/embedding/sagemaker_embedding_driver.py

```diff
@@ -0,0 +1,30 @@
+from __future__ import annotations
+from typing import TYPE_CHECKING
+import json
+from typing import Any
+
+import boto3
```

Comment:
Elsewhere in the framework, like in bedrock_titan_embedding_driver.py, we conditionally `import boto3` under the `if TYPE_CHECKING` block like BaseEmbeddingModelDriver below, but it's not consistent. Should we do that here as well?

Reply:
Good catch, that would've broken things. Also updated the non-typed import of boto3 to use `import_optional_dependency`.